src/jimple-frontend/AST/jimple_expr.cpp

```
@@ -0,0 +1,369 @@
+{
+  std::string expr_type;
+  if(!j.contains("expr_type"))
+  {
+    jimple_constant c;
+    c.setValue("0");
+    return std::make_shared<jimple_constant>(c);
+  }
+
+  j.at("expr_type").get_to(expr_type);
+
+  // TODO: hashmap, the standard is not stable enough yet
+  // It is still a work in progress in the parser: https://github.com/rafaelsamenezes/jimple_parser
+  if(expr_type == "constant")
+  {
+    jimple_constant c;
+    c.from_json(j);
+    return std::make_shared<jimple_constant>(c);
+  }
+
+  if(expr_type == "string_constant")
+  {
+    jimple_constant c;
+    return std::make_shared<jimple_constant>(c);
+  }
+
+  if(expr_type == "symbol")
+  {
+    jimple_symbol c;
+    c.from_json(j);
+    return std::make_shared<jimple_symbol>(c);
+  }
+
+  if(expr_type == "static_invoke")
+  {
+    jimple_expr_invoke c;
+    c.from_json(j);
+    return std::make_shared<jimple_expr_invoke>(c);
+  }
+
+  if(expr_type == "binop")
+  {
+    jimple_binop c;
+    c.from_json(j);
+    return std::make_shared<jimple_binop>(c);
+  }
+
+  if(expr_type == "cast")
+  {
+    jimple_cast c;
+    c.from_json(j);
+    return std::make_shared<jimple_cast>(c);
+  }
+
+  if(expr_type == "lengthof")
+  {
+    jimple_cast c;
+    c.from_json(j);
+    return std::make_shared<jimple_cast>(c);
+  }
+
+  if(expr_type == "newarray")
+  {
+    jimple_newarray c;
+    c.from_json(j);
+    return std::make_shared<jimple_newarray>(c);
+  }
+
+  if(expr_type == "new")
+  {
+    jimple_new c;
+    c.from_json(j);
+    return std::make_shared<jimple_new>(c);
+  }
+
+  if(expr_type == "deref")
+  {
+    jimple_deref c;
+    c.from_json(j);
+    return std::make_shared<jimple_deref>(c);
+  }
+
+  if(expr_type == "nondet")
+  {
+    jimple_nondet c;
+    return std::make_shared<jimple_nondet>(c);
+  }
+
+  if(expr_type == "field_access")
+  {
+    jimple_field_access c;
+    c.from_json(j);
+    return std::make_shared<jimple_field_access>(c);
+  }
+
+  throw "Invalid expr type";
```

In case this is not meant for the JSON library, could we instead throw a proper exception s.t. when not caught at least the message is printed instead of `terminate called after throwing an instance of 'char const*'`?

---

replaced exception with an msg.error + abort